ARGO/src/RenderSystem.cpp

```diff
@@ -0,0 +1,73 @@
+#include "stdafx.h"
+#include "RenderSystem.h"
+
+RenderSystem::~RenderSystem()
+{
+	BaseSystem::~BaseSystem();
+}
+
+void RenderSystem::update(Entity& t_e)
+{
+	//some sort of update to visual components here
+	//maybe shaders/blending?
+}
+
+void RenderSystem::render(SDL_Renderer* t_renderer, Entity& t_e)
+{
+	TransformComponent* posComp = dynamic_cast<TransformComponent*>(t_e.getComponent(ComponentType::Transform));
+	VisualComponent* visComp = dynamic_cast<VisualComponent*>(t_e.getComponent(ComponentType::Visual));
+
+	if (visComp != nullptr)
+	{
+		renderTextures(visComp, posComp->getPos().x, posComp->getPos().y, t_renderer);
+	}
+	else
+	{
+		ColourComponent* colComp = dynamic_cast<ColourComponent*>(t_e.getComponent(ComponentType::Colour));
+		renderPrimitives(t_renderer, posComp, colComp);
+	}
+}
+
+void RenderSystem::renderPrimitives(SDL_Renderer* t_renderer, TransformComponent* t_posComp, ColourComponent* t_colComp)
+{
+	Uint8 prevRGBA[4];
+	SDL_GetRenderDrawColor(t_renderer, &prevRGBA[0], &prevRGBA[1], &prevRGBA[2], &prevRGBA[3]);
+
+	SDL_Rect rect;
+	rect.x = t_posComp->getPos().x - 25;
+	rect.y = t_posComp->getPos().y - 25;
+	rect.w = 50;
+	rect.h = 50;
+
```

Comment:
magic numbers

Reply:
just a size for primitives for the time being